assets/js/modules/analytics-4/components/common/PropertySelect.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * GA4 Property Select component.
+ *
+ * Site Kit by Google, Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { Select, Option } from '../../../../material-components';
+import ProgressBar from '../../../../components/ProgressBar';
+import { STORE_NAME } from '../../datastore/constants';
+import { MODULES_ANALYTICS, PROPERTY_CREATE } from '../../../analytics/datastore/constants';
+import { isValidAccountID } from '../../../analytics/util';
+import { trackEvent } from '../../../../util';
+const { useSelect, useDispatch } = Data;
+
+export default function PropertySelect() {
+	const {
+		accountID,
+		properties,
+		isResolvingProperties,
+	} = useSelect( ( select ) => {
+		const data = {
+			accountID: select( MODULES_ANALYTICS ).getAccountID(),
+			properties: [],
+			isResolvingProperties: false,
+		};
+
+		if ( data.accountID ) {
+			data.properties = select( STORE_NAME ).getProperties( data.accountID );
+			data.isResolvingProperties = select( STORE_NAME ).isResolving( 'getProperties', [ data.accountID ] );
```

Comment:
I think we need to use `hasFinishedResolution` instead of `isResolving`. At least this is what we use across the code base to determine whether something has been resolved or not.

Reply:
Yeah I've not seen this before. It did come from the analytics module's `PropertySelect` component though...

https://github.com/google/site-kit-wp/blob/11b7cff0245e1eae5e3d2ffb085cdb0fceabeef5/assets/js/modules/analytics/components/common/PropertySelect.js#L51

Reply:
Let's use `hasFinishedResolution` since we primarily use it right now.

Reply:
Things seem to break when I do that 🤔

Reply:
> I'm guessing you probably tried replacing isResolving for hasFinishedResolution

I'm unable to recreate the issue now so I think that must have been what was I was doing before, sorry. 🤦‍♂️ 
I've gone with your suggestion below as well Evan, regarding using a single `isLoading` variable. Thanks!